en/22/12.md

```diff
@@ -0,0 +1,277 @@
+
+          #[derive(NestedEncode, NestedDecode, TopEncode, TopDecode, TypeAbi)]
+          pub struct Kitty {
+              pub is_gestating: bool,
+              pub is_ready: bool,
+              pub cooldown_index: u64,
+              pub next_action_at: u64,
+              pub siring_with_id: u64,
+              pub birth_time: u64,
+              pub matron_id: u64,
+              pub sire_id: u64,
+              pub generation: u64,
+              pub genes: u64,
+          }
+
+          #[multiversx_sc::proxy]
+          pub trait CryptoKitties {
+              #[endpoint]
+              fn get_kitty(&self, id: usize) -> Kitty;
+          }
+      }
+
+      #[multiversx_sc::module]
+      pub trait ZombieFeeding: storage::Storage + zombie_factory::ZombieFactory {
+          fn feed_and_multiply(&self, zombie_id: usize, target_dna: u64) {
+              let caller = self.blockchain().get_caller();
+              require!(
+                  caller == self.zombie_owner(&zombie_id).get(),
+                  "Only the owner of the zombie can perform this operation"
+              );
+              let my_zombie = self.zombies(&zombie_id).get();
+              let dna_digits = self.dna_digits().get();
+              let max_dna_value = u64::pow(10u64, dna_digits as u32);
+              let verified_target_dna = target_dna % max_dna_value;
+              let new_dna = (my_zombie.dna + verified_target_dna) / 2;
+              self.create_zombie(caller, ManagedBuffer::from(b"NoName"), new_dna);
+          }
+
+          #[callback]
+          fn get_kitty_callback(
+            &self, 
+            #[call_result] result: ManagedAsyncCallResult<Kitty>,
+            zombie_id: usize
+          ) {
+              match result {
+                  ManagedAsyncCallResult::Ok(kitty) => {
+                    let kitty_dna = kitty.genes;
+                    self.feed_and_multiply(zombie_id, kitty_dna);
+                  },
+                  ManagedAsyncCallResult::Err(_) => {},
+              }
+          }
+
+          #[endpoint]
+          fn feed_on_kitty(
+            &self, 
+            zombie_id: usize,
+            kitty_id: usize,
+          ) {
+            let crypto_kitties_sc_address = self.crypto_kitties_sc_address().get();
+              self.kitty_proxy(crypto_kitties_sc_address)
+                  .get_kitty(kitty_id)
+                  .async_call()
+                  .with_callback(self.callbacks().get_kitty_callback(zombie_id))
+                  .call_and_exit();
+          }
+
+          #[proxy]
+          fn kitty_proxy(&self, to: ManagedAddress) -> crypto_kitties_proxy::Proxy<Self::Api>;
+      }
+---
+We are now ready to complete our `callback`, doing something with the genes of our returned kitty.
+
+
+# Put it to the test
+
+1. Inside the brackets of `ManagedAsyncCallResult::Ok(kitty) => {}` we now need to extract the kitty genes and store them inside a new variable called `kitty_dna`.
```

Comment:
Please consider telling them how to extract the kitty genes, like the actual code they need to write.